### PR TITLE
Appropriate error msg display

### PR DESCRIPTION
**Pull request by aniarya82**

### CODE REVIEW

File: login-complete.php [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?php
  session_start();
  if ($_SESSION["email"] != "") {
-   header('Location: ./signin.php');
+   header('Location: ./signin.php?err=');
```
Comment (icyflame) [inline login-complete.php]: Add the message as

```
You must be logged in to see that page
```

File: login.php [hunk 2]
```diff
@@ -25,12 +25,16 @@
...
+        header('Location: ./signin.php?err=pass');
       }
+    } else {
+      header('Location: ./signin.php?err=email');
     }
   }
 } else {
   $connection->close();
-  header('Location: ./login.php');
+  header('Location: ./signin.php?err=');
```
Comment (icyflame) [inline login.php]: Redirect to `signup.php` and add the error message:

```
Please register before logging in.
```

File: navbar.php [hunk 3]
```diff
@@ -28,7 +28,7 @@
           echo "<li><a href='./index.php'>Logged in as ".$_SESSION['email']."</a></li>";
           echo "<li><a href='./logout.php'>LOGOUT</a></li>";
         } else {
-          echo "<li><a href='./signin.php'>LOGIN</a></li>";
+          echo "<li><a href='./signin.php?err='>LOGIN</a></li>";
```
Comment (icyflame) [inline navbar.php]: What is the `err` parameter here for?

File: signin.php [hunk 4]
```diff
@@ -10,6 +10,30 @@
   <?php include "navbar.php"; ?>
   <div class="container">
     <h2 align="center">Login</h2>
+    <!-- <div class="alert alert-danger alert-dismissible">
```
Comment (icyflame) [inline signin.php]: Remove the commented code.

File: signin.php [hunk 5]
```diff
@@ -10,6 +10,30 @@
...
+      $err = $_GET["err"];
+      echo "<div class='alert alert-danger alert-dismissible'>";
+      echo "<button type='button' class='close' data-dismiss='alert' aria-label='Close'>";
+      echo "<span aria-hidden='true'>&times;</span>";
+      echo "</button>";
+      if ($err == "email") {
+        echo "Email Id not registed";
+      }
+      if ($err == "pass") {
+        echo "Email Id OR Password didn't match";
```
Comment (icyflame) [inline signin.php]: `ID` not `Id`

File: signin.php [hunk 6]
```diff
@@ -10,6 +10,30 @@
...
+    </div> -->
+    <?php
+    if ($_GET["err"]) {
+      $err = $_GET["err"];
+      echo "<div class='alert alert-danger alert-dismissible'>";
+      echo "<button type='button' class='close' data-dismiss='alert' aria-label='Close'>";
+      echo "<span aria-hidden='true'>&times;</span>";
+      echo "</button>";
+      if ($err == "email") {
+        echo "Email Id not registed";
```
Comment (icyflame) [inline signin.php]: `ID` not `Id`

File: connection.php [hunk 7]
```diff
@@ -7,7 +7,8 @@
 $connection = new mysqli($DB_HOST, $DB_USER, $DB_PASS, $DB_NAME);
 
 if ($connection->connect_error) {
-  die("Connecton failed: ".$connection->connect_error);
+  // die("Connecton failed: ".$connection->connect_error);
+  die("It seems that we cannot talk to our DB right now. Please try again in couple of minutes");
```
Comment (icyflame) [inline connection.php]: `in a couple of minutes`

File: login-complete.php (hunk 1 above)
Comment (aniarya82) [inline login-complete.php]: logged in to see which page??

Comment (icyflame) [inline login-complete.php]: The `login-complete.php` page.

File: navbar.php (hunk 3 above)
Comment (aniarya82) [inline navbar.php]: If err is not there in the url the php prompts with an error of undefined variable so i thought to keep it just to remove the error

Comment (icyflame) [inline navbar.php]: Then, check that it is populated using `if($_GET['err'])` and then use, `$_GET['err']`.)

**Comment by icyflame** [comment]
Fixed in 859c786473490986bff66404e7469d2242bb7833

**Pull request closed without merging**